### ISSUE TITLE
How to generate attribute specific images

### ISSUE THREAD

**Issue by sandhyalaxmiK** [rosinality/stylegan2-pytorch#39]
Hi. Thanks for the grate works. 

In https://github.com/Puzer/stylegan-encoder, new  direction in latent space can be obtained for smile, age, gender.
In https://github.com/spiorf/stylegan-encoder given to get latent space directions for more attributes

How can we get latent-space directions for the attributes like pose, smile, age, gender, hair colour,... in stylegan2???

Thanks in advance.

Regards,
Sandhya

**Comment by rosinality** [comment]
I think you can use same approach for stylegan 2.

**Comment by sandhyalaxmiK** [comment]
Hi,

![left_pose_generated](https://user-images.githubusercontent.com/45446006/73913074-1e810b80-48dc-11ea-806e-d6621252c1b1.png)
![left_pose_target](https://user-images.githubusercontent.com/45446006/73913079-204acf00-48dc-11ea-9865-c68f2206e68b.png)
![streight_pose_generated](https://user-images.githubusercontent.com/45446006/73913086-2476ec80-48dc-11ea-915c-65cdf80c5380.png)
![streight_pose_target](https://user-images.githubusercontent.com/45446006/73913092-2640b000-48dc-11ea-99cc-f4c86d492e0d.png)

I have used run_projector.py to generate images for left_pose_target,streight_pose_target.

I am attaching the generated results also.

**Structure-wise getting good result, but features are changing alot for side poses. To get good results which losses are better to add in projector.py**

(I also tried by changing learning rate. But no improvement in results)

Thanks & Regards,
SandhyaLaxmi

**Comment by rosinality** [comment]
I found quality of images that generated by projecting to latent space is limited. You can optimize each style code per layers separately for generate better images. (--w_plus option.)

**Comment by sandhyalaxmiK** [comment]
Thanks for your great patience.  Can you help me of what the exact similar word for w_plus. I have not seen any wlatents like that in stylegan2 projector in tensorflow???

**Comment by sandhyalaxmiK** [comment]
Solved. Thanks
